themes/src/main/resources/theme/base/login/webauthn-authenticate.ftl

```diff
@@ -5,34 +5,61 @@
     <#elseif section = "header">
         ${kcSanitize(msg("webauthn-login-title"))?no_esc}
     <#elseif section = "form">
-
-    <form id="webauth" class="${properties.kcFormClass!}" action="${url.loginAction}" method="post">
-        <div class="${properties.kcFormGroupClass!}">
-            <input type="hidden" id="clientDataJSON" name="clientDataJSON"/>
-            <input type="hidden" id="authenticatorData" name="authenticatorData"/>
-            <input type="hidden" id="signature" name="signature"/>
-            <input type="hidden" id="credentialId" name="credentialId"/>
-            <input type="hidden" id="userHandle" name="userHandle"/>
-            <input type="hidden" id="error" name="error"/>
-        </div>
-    </form>
-
-    <#if authenticators??>
-        <form id="authn_select" class="${properties.kcFormClass!}">
-            <#list authenticators.authenticators as authenticator>
-                <input type="hidden" name="authn_use_chk" value="${authenticator.credentialId}"/>
-            </#list>
-        </form>
-    </#if>
-
-    <form class="${properties.kcFormClass!}">
-        <div class="${properties.kcFormGroupClass!}">
-            <div id="kc-form-buttons" class="${properties.kcFormButtonsClass!}">
-                <input id="authenticateWebAuthnButton" type="button" onclick="webAuthnAuthenticate()" value="${kcSanitize(msg("webauthn-doAuthenticate"))}"
-                   class="${properties.kcButtonClass!} ${properties.kcButtonPrimaryClass!} ${properties.kcButtonBlockClass!} ${properties.kcButtonLargeClass!}">
+        <div id="kc-form-webauthn" class="${properties.kcFormClass!}">
+            <form id="webauth" action="${url.loginAction}" method="post">
+                <input type="hidden" id="clientDataJSON" name="clientDataJSON"/>
+                <input type="hidden" id="authenticatorData" name="authenticatorData"/>
+                <input type="hidden" id="signature" name="signature"/>
+                <input type="hidden" id="credentialId" name="credentialId"/>
+                <input type="hidden" id="userHandle" name="userHandle"/>
+                <input type="hidden" id="error" name="error"/>
+            </form>
+
+            <div class="${properties.kcFormGroupClass!} no-bottom-margin">
+                <#if authenticators??>
+                    <form id="authn_select" class="${properties.kcFormClass!}">
+                        <#list authenticators.authenticators as authenticator>
+                            <input type="hidden" name="authn_use_chk" value="${authenticator.credentialId}"/>
+                        </#list>
+                    </form>
+
+                    <#if shouldDisplayAuthenticators?? && shouldDisplayAuthenticators>
+                        <#if authenticators.authenticators?size gt 1>
+                            <p class="${properties.kcSelectAuthListItemTitle!}">${kcSanitize(msg("webauthn-available-authenticators"))}</p>
```

Comment:
To be honest, I'm not sure about that. I'd rather not to include the title when there's only one registered authenticator in order to keep it simple and don't much bother the user with an additional "unnecessary" info. The opposite way in case with more authenticators and the title included, might inform the user about the fact that the authentication process can use multiple security keys. And in the case, which you've suggested, the title 'Available Security Keys' would be kinda strange, when there's only one authenticator, so we'd need to distinguish those messages by the authenticators count. But I'm not fully convinced about that.

Reply:
Yes, I agree with your original codes. Thanks.